lightly/data/_video.py

```diff
@@ -112,61 +127,125 @@ def read_frame(self, timestamp = None):
         to the right position and then load the frame.
         
         Args:
-            timestamp: Specific timestamp of frame in seconds or None (default: None)
+            timestamp: 
+                Specific timestamp of frame in seconds or None (default: None)
 
         Returns:
             A PIL Image
 
+        Raises:
+            StopIteration:
+                If video is empty or end of video is reached and timestamp is None.
+            ValueError: 
+                If provided timestamp is not in self.timestamps.
+            RuntimeError:
+                If loaded frame has wrong shape.
+
         """
-        if timestamp is not None:
-            if (
-                self.current_timestamp_idx + 1 < len(self.timestamps)
-                and self.timestamps[self.current_timestamp_idx + 1] == timestamp
-            ):
-                # Timestamp is timestamp of next frame. Set current timestamp
-                # index to next frame.
-                self.current_timestamp_idx += 1
+        if not self.timestamps:
+            # Empty video.
+            raise StopIteration()
+
+        if timestamp is None:
+            # Try to read next frame.
+            if self.current_index is None:
+                # Beginning of video.
+                index = 0
+                timestamp = self.timestamps[index]
+            elif self.current_index >= len(self.timestamps):
+                # Reached end of video.
+                raise StopIteration()
```

Comment:
same concerns about raising `StopIteration` as above

Reply:
This only happens if you run `read_frame(None)` and we are at the end of the video. I think in this case we want to raise StopIteration or what would you suggest to do otherwise?